android/src/main/kotlin/com/kasem/media_picker_builder/providers/VideoFileProvider.kt

```diff
@@ -73,28 +74,39 @@ object VideoFileProvider {
                 }
             }
         }
+        mediaMetadataRetriever.release()
     }
 
-    private fun cursorToMediaFile(cursor: Cursor): MediaFile {
-        val fileId = cursor.getLong(0)          //MediaStore.Video.Media._ID
-        val fileDateAdded = cursor.getLong(1)   //MediaStore.Video.Media.DATE_ADDED
-        val filePath = cursor.getString(2)      //MediaStore.Video.Media.DATA
-        val albumName = cursor.getString(3)     //MediaStore.Video.Media.BUCKET_DISPLAY_NAME
-        val albumId = cursor.getLong(4)         //MediaStore.Video.Media.BUCKET_ID
-        val mimeType = cursor.getString(5)      //MediaStore.Video.Media.MIME_TYPE
-        val duration = cursor.getLong(6) / 1000 //MediaStore.Video.Media.DURATION
-        val orientation = cursor.getInt(7)      //MediaStore.Video.Media.ORIENTATION
+
+    private fun cursorToMediaFile(
+            mediaMetadataRetriever: MediaMetadataRetriever,
+            cursor: Cursor
+    ): MediaFile? {
+        val fileId = cursor.getLong(0)
+        val fileDateAdded = cursor.getLong(1)
+        val filePath = cursor.getString(2)
+        val mimeType = cursor.getString(3)
+
+        var duration: Float? = null
+        var orientation: Int = -1
+        try {
+            mediaMetadataRetriever.setDataSource(filePath)
+            duration = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION).toFloat() / 1000
+            orientation = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION).toInt()
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
 
         return MediaFile(
                 fileId,
-                albumId,
+                0,
                 "", //Passing empty string, because real `albumName` was problematic under certain circumstances
                 fileDateAdded,
                 filePath,
                 null,
                 orientation,
                 mimeType,
-                duration,
+                duration?.toDouble(),
```

Comment:
Why not make duration a double at all times?

Reply:
Good point